src/prof_sys.c

```diff
@@ -572,6 +593,16 @@ prof_dump_check_possible_error(prof_dump_arg_t *arg, bool err_cond,
 
 static int
 prof_dump_open_file_impl(const char *filename, int mode) {
+#ifdef _MSC_VER
```

Comment:
What's the difference between` _MSC_VER` and `_WIN32`, should we use the latter to be consistent with the codebase?

Reply:
Both are interchangeable in this case.

- `_MSC_VER` contains the MSVC compiler version (or, in the case of `clang-cl`, the version of MSVC that version of `clang-cl` targets).
- `_WIN32` is set to 1 when targeting x86/x64/arm32/arm64 with either a MSVC compiler, clang-cl, or MinGW's GCC.

Using `_WIN32` is _probably_ more correct here? I'm not sure if MinGW has its own implementation of `creat`, or if it reuses the broken one from the CRT.